### PR TITLE
Update vs code config and provide c_cpp_properties.json with all supported versions

### PR DESCRIPTION
**Pull request by Dragorn421**
Based on what the Makefile passes as -D efines, leaving out only the "make gcc look like a mips compiler for cc_check" flags

### CODE REVIEW

File: docs/c_cpp_properties.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
+{
+    "configurations": [
+        {
+            "name": "N64 oot-gc-us",
```
Comment (AngheloAlf) [inline docs/c_cpp_properties.json:4]: N64 GC? :thonk:

Comment (Dragorn421) [reply docs/c_cpp_properties.json:4]: "N64" as in "cross compiling" or "Not Windows/Linux x86 PC"
any suggestion?

Comment (AngheloAlf) [reply docs/c_cpp_properties.json:4]: I would drop the N64 part entirely
```suggestion
            "name": "oot-gc-us",
```

Comment (fig02) [reply docs/c_cpp_properties.json:4]: fwiw I agree-- I think its fair that the "building for n64" part is implied with this project. For anyone using decomp for anything else, they can change the names

**Comment by cadmic** [comment]
Thanks a lot for adding the new defines, I'll try to keep the up-to-date in the future. Do you plan to switch configurations often? I think I would be a little annoyed having to add new versions and keep all the defines in sync

**Comment by Dragorn421** [comment]
This is not something that absolutely needs to be kept in sync imo, no problem if it rots for a while before an update pass

File: docs/c_cpp_properties.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                "F3DEX_GBI_2",
+                "F3DEX_GBI_PL",
+                "GBI_DOWHILE",
+                "GBI_DEBUG"
+            ],
+            "cStandard": "gnu89"
+        }
+    ],
+    "version": 4
+}
```
Comment (cadmic) [inline docs/c_cpp_properties.json:113]: add a newline?

**Review by cadmic — APPROVED** [review]

File: docs/c_cpp_properties.json (hunk 2 above)
Comment (Dragorn421) [reply docs/c_cpp_properties.json:113]: I can, the vscode json formatter removes it

**Review by fig02 — APPROVED** [review]

File: docs/c_cpp_properties.json (hunk 2 above)
Comment (cadmic) [reply docs/c_cpp_properties.json]: Huh, for me it doesn't remove it (also VS Code) but it's not important

**Pull request merged**